src/examples/zkapps/merkle_tree/merkle_zkapp.ts

```diff
@@ -12,48 +12,41 @@ Merkle Trees give developers the power of storing large amounts of data off-chai
 
 import {
   SmartContract,
-  isReady,
   Poseidon,
   Field,
-  Permissions,
-  DeployArgs,
   State,
   state,
-  CircuitValue,
   PublicKey,
-  UInt64,
-  prop,
   Mina,
   method,
   UInt32,
   PrivateKey,
   AccountUpdate,
   MerkleTree,
   MerkleWitness,
+  Struct,
 } from 'snarkyjs';
 
-await isReady;
-
 const doProofs = true;
 
 class MyMerkleWitness extends MerkleWitness(8) {}
 
-class Account extends CircuitValue {
-  @prop publicKey: PublicKey;
-  @prop points: UInt32;
-
-  constructor(publicKey: PublicKey, points: UInt32) {
-    super(publicKey, points);
-    this.publicKey = publicKey;
-    this.points = points;
-  }
-
+class Account extends Struct({
+  publicKey: PublicKey,
+  points: UInt32,
+}) {
   hash(): Field {
-    return Poseidon.hash(this.toFields());
+    return Poseidon.hash([
+      ...this.publicKey.toFields(),
+      ...this.points.toFields(),
+    ]);
```

Comment:
could we simplify this?

```suggestion
    return Poseidon.hash(Account.toFields(this));
```

Reply:
Wow, my bad. I don't know this exist.